askama/src/filters/mod.rs

```diff
@@ -297,20 +297,53 @@ pub fn trim<T: fmt::Display>(s: T) -> Result<String> {
 
 /// Limit string length, appends '...' if truncated
 #[inline]
-pub fn truncate(s: impl ToString, len: usize) -> Result<String, Infallible> {
-    fn truncate(s: String, len: usize) -> Result<String, Infallible> {
-        let mut s = s.to_string();
-        if s.len() > len {
-            let mut real_len = len;
-            while !s.is_char_boundary(real_len) {
-                real_len += 1;
+pub fn truncate(s: impl fmt::Display, len: usize) -> Result<String> {
```

Comment:
Would it maybe be possible to instead return `Result<Truncator>`, where `Truncator` `impl fmt::Display`, so we 100% avoid the string allocation.

_Also, maybe it's just a coincidence. But I was going to run through the filters, and refactor them into `impl fmt::Display` when possible, after adding `display_some` ;)_

Reply:
Good idea! Will change.

Reply:
Changed.